### ISSUE TITLE
Review 19 pages where ansible_ssh_user is still in use

### ISSUE THREAD

**Issue by AnBuKu** [debops/docs#251]
Since  Ansible 2.0 and higher where `ansible_ssh_user` is deprecated and replaced by `ansible_user`  [tallandtree_]

Search: https://docs.debops.org/en/latest/search.html?q=ansible_ssh_user&check_keywords=yes&area=default

**Comment by AnBuKu** [comment]
Should I re-work the pages one by one? I'll happily do it, up to you :-)

**Comment by ypid** [comment]
I added the substitution to `debops-optimize` so this fix will propagate over time. You can rework roles if you like. Ideally you can ask which roles would be suitable. Maybe checkout `debops.atd` if you like. That one should be easy.

**Comment by AnBuKu** [comment]
Did this one right now: https://github.com/debops/ansible-atd/pull/9

How do I ask? @ypid makes me a list and work on that or I follow this: Search: https://docs.debops.org/en/latest/search.html?q=ansible_ssh_user&check_keywords=yes&area=default

**Comment by ypid** [comment]
https://docs.debops.org/en/latest/search.html?q=ansible_ssh_user&check_keywords=yes&area=default should be ok for this case when you only want to update the variable. I am just not sure if the DebOps Developers could not do this more conveniently when reworking roles anyway. Ideally you would search for all occurances of `ansible_ssh_user` (not just in the docs, but also in the actual `code`).

**Comment by AnBuKu** [comment]
OK, I'll go for your proposal, docs and code + min_ansible_version: '1.9.0' -> min_ansible_version: '2.0' + if needed - name: Debian versions: - wheezy - jessie add - stretch. - Or do I forget about stretch for the moment?

**Comment by ypid** [comment]
When you have tested that the role actually works on stretch why not. I would not just add it just for the sake of it :)

**Comment by AnBuKu** [comment]
True, so I'll leave it in the future (one is already done without testing on stretch) without stretch.

**Comment by AnBuKu** [comment]
@ypid Please, feel free to update/change. If you agree I start to work off the bullet list below.

Next, after [debops/ansible-atd](https://github.com/debops/ansible-atd) are (change to `ansible_user ` and/or to version `2.0.0 as a consequence): 

[debops/ansible-role-ansible](https://github.com/debops/ansible-role-ansible)
  - https://github.com/debops/ansible-role-ansible/blob/master/defaults/main.yml
  - https://github.com/debops/ansible-role-ansible/blob/master/README.md
  - https://github.com/debops/ansible-role-ansible/blob/master/meta/main.yml

[debops/ansible-bootstrap](https://github.com/debops/ansible-bootstrap)
  - https://github.com/debops/ansible-bootstrap/blob/master/defaults/main.yml
  - https://github.com/debops/ansible-bootstrap/blob/master/docs/introduction.rst

[debops/ansible-debops](https://github.com/debops/ansible-debops)
  - https://github.com/debops/ansible-debops/blob/master/defaults/main.yml
  - https://github.com/debops/ansible-debops/blob/master/README.md
  - https://github.com/debops/ansible-debops/blob/master/docs/introduction.rst
  - https://github.com/debops/ansible-debops/blob/master/meta/main.yml

[debops/ansible-kvm](https://github.com/debops/ansible-kvm)
  - https://github.com/debops/ansible-kvm/blob/master/defaults/main.yml
  - https://github.com/debops/ansible-kvm/blob/master/README.md
  - https://github.com/debops/ansible-kvm/blob/master/meta/main.yml

[debops/ansible-librenms](https://github.com/debops/ansible-librenms)
  - https://github.com/debops/ansible-librenms/blob/master/docs/getting-started.rst

[debops/ansible-libvirtd](https://github.com/debops/ansible-libvirtd)
  - https://github.com/debops/ansible-libvirtd/blob/master/defaults/main.yml

[debops/ansible-openvz](https://github.com/debops/ansible-openvz)
  - https://github.com/debops/ansible-openvz/blob/master/defaults/main.yml
  - https://github.com/debops/ansible-openvz/blob/master/README.md
  - https://github.com/debops/ansible-openvz/blob/master/meta/main.yml

[debops/ansible-preseed](https://github.com/debops/ansible-preseed)
  - https://github.com/debops/ansible-preseed/blob/master/defaults/main.yml
  - https://github.com/debops/ansible-preseed/blob/master/README.md
  - https://github.com/debops/ansible-preseed/blob/master/docs/introduction.rst
  - https://github.com/debops/ansible-preseed/blob/master/docs/getting-started.rst
  - https://github.com/debops/ansible-preseed/blob/master/meta/main.yml

[debops/ansible-smstools](https://github.com/debops/ansible-smstools)
  - https://github.com/debops/ansible-smstools/blob/master/defaults/main.yml
  - https://github.com/debops/ansible-smstools/blob/master/README.md
  - https://github.com/debops/ansible-smstools/blob/master/meta/main.yml

**Comment by ypid** [comment]
Looks good to me. But keep in mind that you can automate most of the work when you want to get a little fancy :) Still some changelog stuff and other things might have to be done manually. I thought for a sec that we deprecated `debops.openvz` already but seems it it still supported so go for it :)

**Comment by AnBuKu** [comment]
I'm pretty sure you as a hacker/coder can automate this replacement work.  :-) - But not for me as Joe Average or a simple supporter. :-(

`debops.openvz` is one of 20 roles with status tag `build failing` of [actual 122 DebOps roles]( https://debops.org/status.html). In this status table there is no role with status `deprecated`, maybe as there is no column with status `active` or  `deprecated`.

**Comment by ypid** [comment]
All right.

> In this status table there is no role with status deprecated,

Deprecated disappear from the status page currently. I think there is an issue for status field. It has been requested a number of times but I am not really convinced :)

**Comment by AnBuKu** [comment]
Maybe at the end of the status page, there could be a section `Deprecated roles`? As well, the tag `Deprecated role` would need to be applied in GitHub and in docs.debops.org, short: At any place related to the deprecated role.

Anyhow, for the moment being, I'm not aware of any deprecated role in DebOps.

**Comment by ypid** [comment]
Lets just say I am not convinced that this should be on my todo right now :)

There are some. [`grep` over all roles helps](https://github.com/search?l=YAML&q=org%3Adebops+deprecated&type=Code&utf8=%E2%9C%93).

**Comment by AnBuKu** [comment]
Don't worry, I don't wanna convince you of anything :)

Then following your [`grep` over all roles](https://github.com/search?l=YAML&q=org%3Adebops+deprecated&type=Code&utf8=%E2%9C%93) there are so far two deprecated roles (maybe more?)

- [debops/ansible-mysql](https://github.com/debops/ansible-mysql)
- [debops/ansible-kvm](https://github.com/debops/ansible-kvm)

Still listed in the [DebOps status page](https://debops.org/status.html). Both roles with **Warning, this is a deprecated role** It has been superseded by blabla

**Comment by ypid** [comment]
@drybjed I guess those roles should be removed from debops/docs?

**Comment by AnBuKu** [comment]
Or at least removed from the status page; as long there is no column (Deprecated y/n) or section "Deprecated  roles" at the end of the status page.

Both might make sense, as documentation of such documentation might be asked by [Best Practice Badge](https://bestpractices.coreinfrastructure.org/projects/237) of CII. But I'm not sure about that.

**Comment by ypid** [comment]
Removing from debops/docs will also remove them from the status page.

**Comment by AnBuKu** [comment]
Hmmm... what is your suggestion then?

**Comment by ypid** [comment]
> Hmmm... what is your suggestion then?

Removing from both for now. They can still be found on GitHub.

**Comment by AnBuKu** [comment]
If we do so, I suggest to open an issue "Status page / Deprecated rolls" in github debops/docs as a reminder. Including layout of (far in the) future "status page".

Status page
  |
  |____Active
  |          |____Core rolls
  |          |____Community packages
  |
  |____Deprecated
